### PR TITLE
chore: fix eth gas calc

### PR DESCRIPTION
**Pull request by martinsaporiti**

### CODE REVIEW

File: internal/gateways/publisher_gateway.go [hunk 1]
```diff
@@ -98,6 +98,9 @@ func (pb *PublisherEthGateway) PublishState(ctx context.Context, identifier *w3c
 			log.Error(ctx, "failed to create tx opts", "err", err)
 			return nil, err
 		}
+		log.Info(ctx, "Transaction metadata", "opts.GasPrice:", opts.GasPrice)
+		log.Info(ctx, "Transaction metadata", "opts.GasLimit:", opts.GasLimit)
+		log.Info(ctx, "Transaction metadata", "opts.GasTipCap:", opts.GasTipCap)
```
Comment (x1m3) [inline internal/gateways/publisher_gateway.go:103]:
```suggestion
		log.Info(ctx, "Transaction metadata", "opts.GasPrice:", opts.GasPrice, "opts.GasLimit:", opts.GasLimit, "opts.GasTipCap:", opts.GasTipCap)

```

File: internal/gateways/publisher_gateway.go [hunk 2]
```diff
@@ -131,6 +134,9 @@ func (pb *PublisherEthGateway) PublishState(ctx context.Context, identifier *w3c
...
```
Comment (x1m3) [inline internal/gateways/publisher_gateway.go:140]:
```suggestion
		log.Info(ctx, "Transaction metadata", "opts.GasPrice:", opts.GasPrice, "opts.GasLimit:", opts.GasLimit, "opts.GasTipCap:", opts.GasTipCap)


```

**Review by x1m3 — COMMENTED** [review]
LGTM. Just 2 logs than can be condensed

File: pkg/blockchain/eth/client.go [hunk 3]
```diff
@@ -331,24 +331,36 @@ func (c *Client) CreateTxOpts(ctx context.Context, kmsKey kms.KeyID) (*bind.Tran
...
+	if err != nil {
+		log.Error(ctx, "failed to get gas price", "err", err)
+		return nil, err
+	}
+
+	opts = &bind.TransactOpts{
+		From:     addr,
+		Signer:   sigFn,
+		GasPrice: gasPrice,
+		GasLimit: uint64(c.Config.DefaultGasLimit),
```
[the remark this answers is not in the document]
Comment (martinsaporiti) [reply pkg/blockchain/eth/client.go:361]: I'm following the same approach the library used. If `c.Config.DefaultGasLimit` == 0 the gas limit is estimated.

**Review by ilya-korotya — APPROVED** [review]

**Review by x1m3 — APPROVED** [review]

**Pull request merged**